src/python/gudhi/representations/vector_methods.py

```diff
@@ -123,6 +123,15 @@ def _automatic_sample_range(sample_range, X, y):
                 pass
         return sample_range
 
+
+def trim_on_edges(x, are_endpoints_nan):
```

Comment:
I think the convention would be to call it `_trim_on_edges` with a leading underscore since it is not public.